### ISSUE TITLE
[Bug]: file permission to execute "ss3" downloaded by get_ss3_exe on non windows OS

### ISSUE THREAD

**Issue by yukio-takeuchi** [r4ss/r4ss#842]
### Describe the bug

I am not sure if it is really a bug. I'd rather to make a clarification of the "expected" behavior of get_ss3_exe function.

For  linux and macOS, ss3 downloadable from https://github.com/nmfs-stock-synthesis/stock-synthesis doesn't have file permission to execute while in the case of windows ".exe" extension  immediately allow to "ss3" to be executable.

### To Reproduce

On macOS or linux

```
➜ Rscript -e "r4ss::get_ss3_exe()"
No directory provided, the executable will be downloaded to the
            working directory
trying URL 'https://github.com/nmfs-stock-synthesis/stock-synthesis/releases/download/v3.30.21/ss_osx'
Content type 'application/octet-stream' length 4139312 bytes (3.9 MB)
==================================================
downloaded 3.9 MB

The stock synthesis executable for Mac v3.30.21 was downloaded to: /Users/yukiot/test_get_ss3_exe/ss3

~/test_get_ss3_exe took 3s
➜ ./ss3
zsh: permission denied: ./ss3
❯ ls -lh ss3
-rw-r--r--  1 yukiot  staff   3.9M Jul  8 15:55 ss3
```

### Expected behavior

If it is better to make "ss3" file executable immediately after download on linux and macOS, it may be better to add an additional code to change file permission in the case of linux or macOS.

If so 

a line 

```
Sys.chmod(paths = file.path(dir, "ss3"), mode =  "0700")
```

may work to make "ss3" executable.

But alternative option may be to include an addtional argument of get_ss3_exe to specify the desirable file permission of "ss3", e.g. "mode = '0777'". Then a new line like

```
Sys.chmod(paths = file.path(dir, "ss3"), mode =  mode)
```

### Which OS are you seeing the problem on?

Mac, Linux

### Which versions of r4ss and SS3 are you seeing the problem on?

latest r4ss@main

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/r4ss/r4ss#code-of-conduct)

**Comment by e-gugliotti-NOAA** [comment]
That is a great point @yukio-takeuchi. I will get working on making the exe immediately usable to linux and max users today.

**Comment by iantaylor-NOAA** [comment]
Fixed by @e-gugliotti-NOAA via #844.
